crates/configuration/src/lib.rs

```diff
@@ -0,0 +1,266 @@
+use anyhow::{anyhow, Result};
+use std::fs;
+
+use camino::Utf8PathBuf;
+use tempfile::{tempdir, TempDir};
+use toml::Value;
+pub const CONFIG_FILENAME: &str = "snfoundry.toml";
+
+/// Defined in snfoundry.toml
+/// Configuration not associated with any specific package
+pub trait GlobalConfig {
+    #[must_use]
+    fn tool_name() -> &'static str;
+
+    fn from_raw(config: &Value) -> Result<Self>
+    where
+        Self: Sized;
+}
+
+/// Defined in scarb manifest
+/// Configuration associated with a specific package
+pub trait PackageConfig {}
+
+pub fn get_profile(raw_config: &Value, tool: &str, profile: &Option<String>) -> Result<Value> {
+    let profile_name = profile.as_deref().unwrap_or("default");
+    let config = raw_config
+        .get(tool)
+        .expect("Failed to find sncast config in snfoundry.toml file");
```

Comment:
probably a nitpick, since firge does not use snfoundry.toml yet but since we're in a shared crate, shouldn't this be generalised?

Reply:
It should, I missed this one :0

Reply:
Done